public/js/main.js

```diff
@@ -291,11 +298,32 @@ function setFileURLs(new_provider) {
       scrollProgress.update();
     } else if (lastQuery !== val) {
       animateTop();
-      index.search(val, displayMatchingLibraries);
+      displayItems = 20;
+      queryItems = 100;
+      lazyScroll = true;
+      index.search(val, {hitsPerPage: queryItems}, function(err, content){
+        cachedQueryResult = content;
+        displayMatchingLibraries(err, content);
+      });
     }
     lastQuery = val;
   }
 
+  var windowSelector = $(window);
+  windowSelector.scroll(_.debounce( function() {
+    if (lazyScroll && windowSelector.scrollTop() + windowSelector.height() * 2 >= $(document).height()) {
+      displayItems += 20;
+      if (displayItems > queryItems) {
+        queryItems += 100;
```

Comment:
I blindly assume that we caches 100 results.
Or maybe this number should controlled by numbers of hits.